test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,454 @@
+            entity_value
+                .get("version_major")
+                .and_then(|entity_value| entity_value.as_str())
+        }) {
+            if let Ok(version_major_parsed) = entity_value.parse::<u32>() {
+                entity.version_major = Some(version_major_parsed);
+            }
+        } else {
+            error!("Error: entity_value version major is not a string");
+        };
+
+        if let Some(entity_value) = value.get("entity").and_then(|entity_value| {
+            entity_value
+                .get("version_minor")
+                .and_then(|entity_value| entity_value.as_str())
+        }) {
+            if let Ok(version_minor_parsed) = entity_value.parse::<u32>() {
+                entity.version_minor = Some(version_minor_parsed);
+            }
+        } else {
+            error!("Error: entity version minor is not a string");
+        };
+
+        entity.special_fields = SpecialFields::default();
+
+        if let Some(resource_value) = value
+            .get("resource")
+            .and_then(|resource_value| resource_value.get("name"))
+            .and_then(|resource_value| resource_value.as_str())
+        {
+            resource.name = resource_value.to_owned();
+        } else {
+            error!("Error: Name field is not a string in resource");
+        };
+
+        if let Some(resource_value) = value
+            .get("resource")
+            .and_then(|resource_value| resource_value.get("instance"))
+            .and_then(|resource_value| resource_value.as_str())
+        {
+            resource.instance = Some(resource_value.to_owned());
+        } else {
+            error!("Error: instance field is not a string in resource");
+        }
+
+        if let Some(resource_value) = value
+            .get("resource")
+            .and_then(|resource_value| resource_value.get("message"))
+            .and_then(|resource_value| resource_value.as_str())
+        {
+            resource.message = Some(resource_value.to_owned());
+        } else {
+            error!("Error: message field is not a string in resource_value");
+        };
+
+        if let Some(resource_value) = value
+            .get("resource")
+            .and_then(|resource_value| resource_value.get("id"))
+            .and_then(|resource_value| resource_value.as_str())
+        {
+            if let Ok(parsed_id) = resource_value.parse::<u32>() {
+                resource.id = Some(parsed_id);
+            } else {
+                error!("Error: id field parsing to u32");
+            }
+        } else {
+            error!("Error: id field is not string");
+        };
+
+        if !(authority.get_name().is_none() && authority.number.is_none()) {
+            dbg!(" authority is not default");
+            uuri.authority = MessageField(Some(Box::new(authority)));
+        }
+        uuri.entity = MessageField(Some(Box::new(entity)));
+        uuri.resource = MessageField(Some(Box::new(resource)));
```

Comment:
Suggest re-reading the recommendations I gave last time.
* try to avoid mutation
* use the `let foo = ...` construct as suggested
* construct each component (UAuthority, UEntity, UResource, UUri) one at a time e.g. as suggested [in this comment](https://github.com/eclipse-uprotocol/up-tck/pull/23#discussion_r1566166247)